test-utils/runtime-tester/src/fuzzing.rs

```diff
@@ -184,19 +391,177 @@ impl Scope {
         self.nonce
     }
 
+    pub fn num_alive_accounts(&self) -> usize {
+        self.alive_accounts.len()
+    }
+
+    pub fn random_non_zero_alive_account_usize_id(&self, u: &mut Unstructured) -> Result<usize> {
+        let mut accounts = self.alive_accounts.clone();
+        accounts.remove(&0);
+        Ok(*u.choose(&accounts.iter().cloned().collect::<Vec<_>>())?)
+    }
+
+    pub fn random_alive_account_usize_id(&self, u: &mut Unstructured) -> Result<usize> {
+        Ok(*u.choose(&self.alive_accounts.iter().cloned().collect::<Vec<_>>())?)
+    }
+
     pub fn random_account(&self, u: &mut Unstructured) -> Result<Account> {
-        Ok(u.choose(&self.accounts)?.clone())
+        Ok(self.accounts[self.random_alive_account_usize_id(u)?].clone())
+    }
+
+    pub fn random_non_zero_account(&self, u: &mut Unstructured) -> Result<Account> {
+        Ok(self.accounts[self.random_non_zero_alive_account_usize_id(u)?].clone())
     }
 
     pub fn new_account(&mut self) -> Account {
         let new_id = format!("test{}", self.accounts.len());
-        self.accounts.push(Account::from_id(AccountId::try_from(new_id).unwrap()));
+        self.alive_accounts.insert(self.accounts.len());
+        self.accounts.push(Account::from_id(new_id));
         self.accounts[self.accounts.len() - 1].clone()
     }
+
+    pub fn delete_account(&mut self, account_usize_id: usize) {
+        self.alive_accounts.remove(&account_usize_id);
+    }
+
+    pub fn deploy_contract(&mut self, receiver_account: &Account, contract_id: usize) {
+        let acc_id = receiver_account.usize_id();
+        self.accounts[acc_id].deployed_contracts.insert(contract_id);
+    }
 }
 
 impl Account {
-    pub fn from_id(id: AccountId) -> Self {
-        Self { id, balance: TESTING_INIT_BALANCE }
+    pub fn from_id(id: String) -> Self {
+        Self {
+            id: AccountId::from_str(id.as_str()).expect("Invalid account_id"),
+            balance: TESTING_INIT_BALANCE,
+            deployed_contracts: HashSet::default(),
+        }
+    }
+
+    pub fn usize_id(&self) -> usize {
+        self.id.as_ref()[4..].parse::<usize>().unwrap()
+    }
+}
+
+impl Function {
+    pub fn arbitrary(&self, u: &mut Unstructured) -> Result<FunctionCallAction> {
+        match self {
+            Function::StorageUsage => Ok(FunctionCallAction {
+                method_name: "ext_storage_usage".to_string(),
+                args: vec![],
+                gas: GAS_1,
+                deposit: 0,
+            }),
+            Function::BlockIndex => Ok(FunctionCallAction {
+                method_name: "ext_block_index".to_string(),
+                args: vec![],
+                gas: GAS_1,
+                deposit: 0,
+            }),
```

Comment:
One way to compress such "most cases are identical, but few are not like the others" is to create a default and modify it:

```
let mut res =  FunctionCallAction {
                method_name: String::new(),
                args: vec![],
                gas: GAS_1,
                deposit: 0,
            };
match self {
  Function::StorageUsage => res.method_name = "ext_storage_usage".to_string();
  ...
}
Ok(res)
```